File: Data_Structures/python/Linked_lists/removeDuplication.py
class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if head is None or head.next is None: return head

        fp, sp = head, head.next

        while sp is not None:
            if fp.val == sp.val:
                fp.next = fp.next.next
            else:
                fp = fp.next
            sp = sp.next
        return head

File: cracking_the_coding_interview/chapter-two-linked-lists/removeDuplication.js
const removeDuplication = head => {
    let tmp = null;
    let ptr2 = null;
    let ptr1 = head;
    while (ptr1 && ptr1.next) {
        ptr2 = ptr1;
        while (ptr2.next) {
            if (ptr1.val == ptr2.val) {
                tmp = ptr1.next;
                ptr1.next = ptr2.next.next;
                delete(tmp);
            } else {
                ptr2 = ptr2.next;
            }
        }
        ptr1 = ptr1.next;
    }
}
